src/Control/Exception/Safe.hs

```diff
@@ -172,7 +172,9 @@ evaluateDeep action = do
 catchAnyDeep :: (C.MonadCatch m, MonadIO m, NFData a) => m a -> (SomeException -> m a) -> m a
 catchAnyDeep = catchDeep
 
--- | 'catch' without async exception safety
+-- | 'catch' without async exception safety.
+--
+-- Will catch extension types which are children of 'Control.Exception.SomeAsyncException', see <https://github.com/fpco/safe-exceptions#determining-sync-vs-async>
```

Comment:
`catchAsync` can in fact catch any exception type, consider:

```haskell
import Control.Concurrent (myThreadId)
import Control.Exception (ArithException (DivideByZero))
import Control.Exception.Safe

main :: IO ()
main = do
    self <- myThreadId
    catchAsync
        (throwTo self DivideByZero)
        (\e -> print ("got an exception", e :: SomeException))
```

The issue is that, when using `throwTo`, a synchronously typed exception is wrapped up in an `AsyncExceptionWrapper`. You can see this too with:

```haskell
import Control.Concurrent (myThreadId)
import Control.Exception (ArithException (DivideByZero))
import Control.Exception.Safe

main :: IO ()
main = do
    self <- myThreadId
    catchAsync
        (throwTo self DivideByZero)
        (\(AsyncExceptionWrapper e) ->
            case fromException $ toException e of
                Just DivideByZero -> putStrLn "caught DivideByZero"
                _ -> print ("something else", e))
```